rpc/executables.go

```diff
@@ -0,0 +1,155 @@
+	// can't use felt.Felt here because prime is larger than felt
+	Prime                  string          `json:"prime"`
+	CompilerVersion        string          `json:"compiler_version"`
+	Bytecode               []*felt.Felt    `json:"bytecode"`
+	Hints                  json.RawMessage `json:"hints"`
+	BytecodeSegmentLengths []int           `json:"bytecode_segment_lengths,omitempty"`
+}
+
+func (h *Handler) CompiledCasm(classHash *felt.Felt) (*CasmCompiledContractClass, *jsonrpc.Error) {
+	state, stateCloser, err := h.bcReader.HeadState()
+	if err != nil {
+		return nil, jsonrpc.Err(jsonrpc.InternalError, err.Error())
+	}
+	defer h.callAndLogErr(stateCloser, "failed to close state reader")
+
+	declaredClass, err := state.Class(classHash)
+	if err != nil {
+		return nil, jsonrpc.Err(jsonrpc.InternalError, err.Error())
+	}
+
+	switch class := declaredClass.Class.(type) {
+	case *core.Cairo0Class:
+		resp, err := adaptCairo0Class(class)
+		if err != nil {
+			return nil, jsonrpc.Err(jsonrpc.InternalError, err.Error())
+		}
+		return resp, nil
+	case *core.Cairo1Class:
+		return adaptCompiledClass(class.Compiled), nil
+	}
+
+	return nil, jsonrpc.Err(jsonrpc.InternalError, "unsupported class type")
+}
+
+func adaptCairo0Class(class *core.Cairo0Class) (*CasmCompiledContractClass, error) {
+	program, err := utils.Gzip64Decode(class.Program)
+	if err != nil {
+		return nil, err
+	}
+
+	var cairo0 zero.ZeroProgram
+	err = json.Unmarshal(program, &cairo0)
+	if err != nil {
+		return nil, err
+	}
+
+	var bytecode []*felt.Felt
+	for _, str := range cairo0.Data {
+		f, err := new(felt.Felt).SetString(str)
+		if err != nil {
+			return nil, err
+		}
+		bytecode = append(bytecode, f)
+	}
+
+	classHints, err := hintRunnerZero.GetZeroHints(&cairo0)
+	if err != nil {
+		return nil, err
+	}
+
+	var hints [][2]any // slice of 2-element tuples where first value is pc, and second value is slice of hints
+	for pc, hintItems := range utils.OrderMap(classHints) {
+		hints = append(hints, [2]any{pc, hintItems})
+	}
+	rawHints, err := json.Marshal(hints)
+	if err != nil {
+		return nil, err
+	}
+
+	adaptEntryPoint := func(ep core.EntryPoint) CasmEntryPoint {
+		return CasmEntryPoint{
+			Offset:   ep.Offset,
+			Selector: ep.Selector,
+			Builtins: nil,
+		}
+	}
+
+	result := &CasmCompiledContractClass{
+		EntryPointsByType: EntryPointsByType{
+			Constructor: utils.Map(class.Constructors, adaptEntryPoint),
+			External:    utils.Map(class.Externals, adaptEntryPoint),
+			L1Handler:   utils.Map(class.L1Handlers, adaptEntryPoint),
+		},
+		Prime:                  cairo0.Prime,
+		Bytecode:               bytecode,
+		CompilerVersion:        cairo0.CompilerVersion,
+		Hints:                  json.RawMessage(rawHints),
+		BytecodeSegmentLengths: nil, // Cairo 0 classes don't have this field (it was introduced since Sierra 1.5.0)
+	}
+	return result, nil
+}
+
+func adaptCompiledClass(class *core.CompiledClass) *CasmCompiledContractClass {
```

Comment:
Maybe it's obvious to everybody but it takes me a few seconds to recall - so please add a comment it's for cairo1 classes (feel free to ignore)

Reply:
I think it's ok without comment. The only way to get a compiled class is to pass corresponding field from cairo1 class